src/macros.rs

```diff
@@ -7,41 +7,12 @@
 //!
 //! This macro moves all of that complexity to build-time. At least for the SPIR-V part of the
 //! shader pipeline. (`gfx-hal` backends have their own SPIR-V-to-native compilers at runtime.)
-//!
-//! Cribbed with permission from Ralith
-//! See: https://github.com/MaikKlein/ash/pull/245
 
-/// Include correctly aligned and typed precompiled SPIR-V
-///
-/// Does not account for endianness mismatches between the SPIR-V file and the target. See
-/// [`wgpu::read_spirv`] for a more general solution.
 #[macro_export]
 macro_rules! include_spv {
-    ($path:expr) => {
-        &$crate::Align4(*include_bytes!($path)) as &$crate::Spirv
-    };
+    ($path:expr) => {{
+        let arr = include_bytes!($path).to_vec();
```

Comment:
I think it will work without `.to_vec()`. You may need to use `&*arr` below.

Reply:
I didn"t get it to work because `std::io::Cursor` needs T to implement `AsRef<[u8]>` and this trait is only implemented for arrays up to a size of 32. Maybe with const generics this will be possible but they havent landed yet.

Reply:
By doing `&*arr` or `&arr[...]`, the array will turn inti a slice, which implements `AsRef<[u8]>`.

Reply:
this still wont work because `AsRef<[u8]>` is only implemented for arrays of sizes up to 32.

I'm getting this error when i do what you suggested:
```
error[E0277]: arrays only have std trait implementations for lengths 0..=32
   --> src/macros.rs:15:26
    |
15  |         wgpu::read_spirv(std::io::Cursor::new(&*arr))
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::array::LengthAtMost32` is not implemented for `[u8; 1484]`
```

Reply:
ah ok now I see what you mean. it works with `&arr[..]`. I thought it was the same as `&*arr`

Reply:
Since the slice is getting copied into Vec<T> anyway, I don’t see a reason not to use Cursor<Vec<u8>>, which satisfies all of the requirements for wgpu::read_spirv

I know it will possibly do more copies and use more heap. But my primary motivation for the macro is to avoid reading from FS at runtime.

Reply:
Yeah but why take that unnecessary allocation if you can just avoid it that easily.
My latest commit addresses this and also merged changes from master.

Reply:
The most recent patch looks great. Can you also move the borrow inside the macro? Just do we don't have to write `&include_spv!(...)`

Then I think this will be perfect.